docs/audio.rst

```diff
@@ -83,8 +86,216 @@ The built-in sounds can be called using ``audio.play(Sound.NAME)``.
 * ``Sound.TWINKLE``
 * ``Sound.YAWN``
 
+Sounds Example
+--------------
+
+::
+
+    from microbit import *
+
+    while True:
+        if button_a.is_pressed() and button_b.is_pressed():
+            # When pressing both buttons only play via the edge connector
+            audio.play(Sound.HELLO, pin=pin0)
+        elif button_a.is_pressed():
+            # On button A play a sound and when it's done show an image
+            audio.play(Sound.HAPPY)
+            display.show(Image.HAPPY)
+        elif button_b.is_pressed():
+            # On button B play a sound and show an image at the same time
+            audio.play(Sound.TWINKLE, wait=False)
+            display.show(Image.BUTTERFLY)
+
+        sleep(500)
+        display.clear()
+
+
+Sound Effects **V2**
+====================
+
+.. py:class::
+    SoundEffect(preset=None, freq_start=500, freq_end=2500, duration=500, vol_start=255, vol_end=0, wave=WAVE_SQUARE, fx=None, interpolation=INTER_LOG)
+
+    An ``SoundEffect`` instance represents a sound effect, composed by a set of
+    parameters configured via the constructor or attributes.
+
+    All the parameters are optional, with default values as shown above, and
+    they can all be modified via attributes of the same name. For example, we
+    can first create an effect ``my_effect = SoundEffect(duration=1000)``,
+    and then change its attributes ``my_effect.duration = 500``.
+
+    :param preset: An existing SoundEffect instance to use as a base, its values
+        are cloned in the new instance, and any additional arguments provided
+        overwrite the base values.
```

Comment:
The interaction with a preset and default keyword arguments is problematic.

Consider `SoundEffect()`.  Does that create a new instance with all default values?

Consider `SoundEffect(vol_end=255)`.  Does that create a new instance with all default values except `vol_end` which is overridden to 255?

Consider `SoundEffect(audio.SoundEffect.CLICK)`.  Does that copy `CLICK` and then override everything with the default values?  If not, how do you explain the difference between that and `SoundEffect()` behaviour?

-----

How about changing the constructor signature to:
```python
SoundEffect(preset=SoundEffect.DEFAULT, *, freq_start, freq_end, duration, vol_start, vol_end, wave, fx, interpolation)
```
Then `SoundEffect.DEFAULT` has the default values for the keyword-only args as previously specified above (freq_start=500 etc).

Reply:
> Consider SoundEffect(). Does that create a new instance with all default values?

Yes, I think this one is straight forward.

> Consider SoundEffect(vol_end=255). Does that create a new instance with all default values except vol_end which is overridden to 255?

Yes, as it would if the `preset` argument didn't exist.


> Consider `SoundEffect(audio.SoundEffect.CLICK)`. Does that copy `CLICK` and then override everything with the default values? If not, how do you explain the difference between that and `SoundEffect()` behaviour?

Okay, I could see how maybe that is a bit less obvious. 
The way I would see it is that `SoundEffect(audio.SoundEffect.CLICK)` is like a copy constructor, so it ignores any of the `__init__` default values.
I think `SoundEffect(audio.SoundEffect.CLICK, duration=2000)` could be easily read as "copy all `audio.SoundEffect.CLICK` values, but overwrite `duration` to `2000`, but I could see how that might not be as obvious if written as  `SoundEffect(duration=2000, preset=audio.SoundEffect.CLICK)`.

>Then SoundEffect.DEFAULT has the default values for the keyword-only args as previously specified above (freq_start=500 etc).

The disadvantage of that is we wouldn't be able to created shorter strings for serialisation with positional arguments.
So this shorter string wouldn't eval:
```
SoundEffect(988, 440, 190, 255, 255, 0, 1, 8)
```

If the confusion arising from having a `preset` argument in the constructor, what if the only way to duplicate and modify would be to use a `copy()` method?
```
my_effect = audio.SoundEffect.CLICK.copy()
my.effect.duration = 2000
```
It's an extra line, but might be easier to understand?

Reply:
> SoundEffect(988, 440, 190, 255, 255, 0, 1, 8)

This won't work because 988 is not a valid preset.  It would need to be `SoundEffect(None, 988, 440, 190, 255, 255, 0, 1, 8)`.

Actually, the way it's implemented at the moment is that if you pass `None` as the preset argument then the preset is the default set of values.  And any extra keyword args override that default /None preset value.

The signature for that is:
```python
SoundEffect(preset=None, *, freq_start, freq_end, duration, vol_start, vol_end, wave, fx, interpolation)
```

Reply:
> > SoundEffect(988, 440, 190, 255, 255, 0, 1, 8)
> 
> This won't work because 988 is not a valid preset. It would need to be `SoundEffect(None, 988, 440, 190, 255, 255, 0, 1, 8)`.

Yes, sorry, that would only work like that if we moved `preset` to be the last parameter, as it was suggested in https://github.com/bbcmicrobit/micropython/pull/753#discussion_r911437405. I thought I had added a comment somewhere to indicate that having it as the last parameter would help with `__repr__`, but I couldn't find it, so I probably forgot to write it down.

-----

The more I think about it, the better the `copy()` method sounds. Do you have any thoughts about it?

Reply:
It's been agreed to remove the `preset` constructor parameter, and add a `copy()` method to be able to clone SoundEffects.